exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/h2/H2Tests.kt

```diff
@@ -69,6 +78,25 @@ class H2Tests : DatabaseTestsBase() {
         }
     }
 
+    @Test
+    fun closeAndUnregister() {
+        withDb(TestDB.H2) { testDB ->
+            runBlocking {
+                val db = requireNotNull(testDB.db) { "testDB.db cannot be null" }
+                TransactionManager.registerManager(
```

Comment:
I'm not 100% sure anymore why this introduced the issue.
I'd need to re-debug the flow, but it was related to `registerManager` adding a 2nd entry for `Database` into ` private val databases = ConcurrentLinkedDeque<Database>()`.

So is that second instance leaking in the `ConcurrentLinkedDeque` if you only call `closeAnddUnregister`? 🤔